### ISSUE TITLE
systemd-nspawn@.service incorrectly enabled (no containers configured on the machine)

### ISSUE THREAD

**Issue by abderrahim** [systemd/systemd#8755]
### Submission type

 - Bug report

### systemd version the issue has been seen with

> v237 (46659f7deb962f55c728e70597e37c2a3ab6326d)

### Used distribution

> [freedesktop-sdk](https://gitlab.com/freedesktop-sdk/freedesktop-sdk)

### In case of bug report: Unexpected behaviour you saw

> On first boot, with no container setup, I see this error
> ```
> [FAILED] Failed to start Container machines.
> See 'systemctl status systemd-nspawn@machines.service' for details.
> ```

### In case of bug report: Steps to reproduce the problem

> The problem is caused by the file `/usr/etc/systemd/system/machines.target.wants/systemd-nspawn@.service`, which is installed because of the following two lines at the end of `/usr/lib/systemd/system/systemd-nspawn@.service`
>```
>[Install]
>WantedBy=machines.target
>```

I don't think it does make sense to install services ending in @ (I don't know how they're called) by default. Either the above two lines should be removed or systemd should be smarter and not symlink them without arguments.

**Comment by poettering** [comment]
uh?

systemd won't enable this unit for you just like that. Which tool did?

the [Install] section in systemd-nspawn@.service exists so that people can invoke "systemctl enable systemd-nspawn@foo.service" and this will hook a new container instance "foo" into the boot just like that...

I think there's some misunderstanding there, what this all does. Can you explain how exactly you ran into this?

**Comment by abderrahim** [comment]
I have no idea what created it, except it didn't exist on the filesystem image before I started it in qemu, and it is there when I log in after the first boot. (And I really mean first boot as in `ConditionFirstBoot=yes`)

before
```
/usr/etc/systemd/system:
dbus-org.freedesktop.network1.service
dbus-org.freedesktop.resolve1.service
getty.target.wants
multi-user.target.wants
network-online.target.wants
sockets.target.wants
sysinit.target.wants

/usr/etc/systemd/system/getty.target.wants:
getty@tty1.service

/usr/etc/systemd/system/multi-user.target.wants:
machines.target
remote-fs.target
systemd-networkd.service
systemd-resolved.service

/usr/etc/systemd/system/network-online.target.wants:
systemd-networkd-wait-online.service

/usr/etc/systemd/system/sockets.target.wants:
systemd-networkd.socket

/usr/etc/systemd/system/sysinit.target.wants:
systemd-timesyncd.service

```

after
```
/usr/etc/systemd/system:
ctrl-alt-del.target
dbus-org.freedesktop.network1.service
dbus-org.freedesktop.resolve1.service
getty.target.wants
machines.target.wants
multi-user.target.wants
network-online.target.wants
sockets.target.wants
sysinit.target.wants

/usr/etc/systemd/system/getty.target.wants:
getty@tty1.service
serial-getty@.service

/usr/etc/systemd/system/machines.target.wants:
systemd-nspawn@.service

/usr/etc/systemd/system/multi-user.target.wants:
machines.target
remote-fs.target
systemd-networkd.service
systemd-resolved.service

/usr/etc/systemd/system/network-online.target.wants:
systemd-networkd-wait-online.service

/usr/etc/systemd/system/sockets.target.wants:
systemd-networkd.socket

/usr/etc/systemd/system/sysinit.target.wants:
systemd-timesyncd.service

```

**Comment by filbranden** [comment]
I'm wondering if this might have been created by someone running `systemctl enable systemd-nspawn@.service` which is allowed but doesn't really make sense:

```shell
$ sudo systemctl enable systemd-nspawn@.service
Created symlink /etc/systemd/system/machines.target.wants/systemd-nspawn@.service → /usr/lib/systemd/system/systemd-nspawn@.service.
$ systemctl --version
systemd 238
+PAM +AUDIT +SELINUX +IMA -APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD +IDN2 -IDN +PCRE2 default-hierarchy=hybrid
```

Maybe we should check for that and block it in `systemctl enable`?

Cheers,
Filipe

**Comment by abderrahim** [comment]
A couple ideas:
- When I type `systemctl status systemd-nspawn@machines.service` as instructed, I see `vendor preset: enabled`. What defines the "vendor preset" variable?
- Is there a way to find out what created this? I see the same behaviour if I use `systemd-nspawn -b` instead of qemu if that's easier to debug.

**Comment by SjonHortensius** [comment]
* vendor-presets are defined in `/usr/lib/systemd/system-preset/*.preset`, see `man systemd.preset`
* Does `ls -al /etc/systemd/system/machines.target.wants/` give you a hint of the time when this unit was enabled?

**Comment by poettering** [comment]
This was fixed by #10954 I guess. Closing.
